core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -174,6 +177,24 @@ private[spark] class CoarseGrainedExecutorBackend(
 
 private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
+  private def addMesosDelegationTokens(driverConf: SparkConf) {
+    val value = driverConf.get("spark.mesos.kerberos.userCredentials")
```

Comment:
You really should be using the config constant you create here.

But a bigger issue is that this value will be written to event logs and shown in the UI. Instead of using the config for this, would it be too hard to add this as a field in `SparkAppConfig` (see `CoarseGrainedClusterMessage.scala`)?